tests/MongoDB.Driver.Core.TestHelpers/CoreTestConfiguration.cs

```diff
@@ -42,8 +42,8 @@ public static class CoreTestConfiguration
         #region static
         // static fields
         private static Lazy<ICluster> __cluster = new Lazy<ICluster>(CreateCluster, isThreadSafe: true);
-        private static ConnectionString __connectionString = GetConnectionString();
-        private static DatabaseNamespace __databaseNamespace = GetDatabaseNamespace();
+        private static Lazy<ConnectionString> __connectionString = new Lazy<ConnectionString>(GetConnectionString, isThreadSafe: true);
+        private static Lazy<DatabaseNamespace> __databaseNamespace = new Lazy<DatabaseNamespace>(GetDatabaseNamespace, isThreadSafe: true);
```

Comment:
Use Lazy because it's better not to throw exceptions from static constructors.

We prefer any exceptions to be thrown later when the corresponding properties are first referenced.